README.md

```diff
@@ -1 +1,112 @@
-TODO
+# WhyLabs Toolkit
+
+The WhyLabs Toolkit package contains helper methods to help users interact with our internal APIs. Users will benefit from using it if they want to abstract some of WhyLabs' internal logic and also automate recurring API calls.
+
+## Configuration
+In order to configure `whylabs_toolkit`, you will need to set `WHYLABS_API_KEY` as an environment variable. With that, the package will be able to authenticate with WhyLabs' API endpoints. You can configure a token for your account directly on the platform. 
+
+To manage dependencies we use [Poetry](https://python-poetry.org/) and also a handful of `Makefile` commands. To install all necessary dependencies and activate the virtual environment, run:
+
+```bash
+make setup && poetry shell
+```
+
+## Usage
+In here we will describe some examples on how to use the package. You can also check [the integration tests directory](./tests/) to have more insights on how things are built to be used.
+### Models
+Users can change their model type between `REGRESSION`, `CLASSIFICATION` and `EMBEDDINGS`, using the models helpers, as the example shows:
+```python
+from whylabs_toolkit.helpers.models import update_model_metadata
+
+update_model_metadata(
+    org_id="org_id",
+    dataset_id="dataset_id",
+    model_type="CLASSIFICATION"
+)
+```
+
+And to change the model granularity:
+
+```python
+update_model_metadata(
+    dataset_id="dataset_id", 
+    org_id="org_id", 
+    time_period="P1M"
+)
+```
+
+>**NOTE**: Learn more on the time period config options with the `whylabs_client.model.time_period.TimePeriod` class, available after you've configured your environment with the described make command above.
+
+### Entity Schema
+Entity Schema helpers assist users to change some of their dataset metadata, such as data types, discreteness and column classification (between inputs and outputs). Here's an example that covers all three cases:
+
+#### Column Classes
+```python
+from whylabs_toolkit.helpers.schema import (
+    UpdateColumnClasses, 
+    UpdateEntityDataTypes, 
+    ColumnsClasses,
+    UpdateColumnsDiscreteness,
+    ColumnsDiscreteness
+)
+
+# Note that you don't need to specify all existing columns, but only those you wish to modify
+
+classes = ColumnsClasses(
+    inputs=[],
+    outputs=["actual_temperature", "predicted_temperature"]
+)
+
+update_entity = UpdateColumnClasses(
+    classes=classes,
+    dataset_id="dataset_id",
+    org_id="org_id"
+)
+
+update_entity.update()
+
+```
+#### Data types
+```python
+columns_schema = {
+    "temperature": "fractional",
```

Comment:
how do we know what datatypes are supported? THe values should be enums here

Reply:
the values are validated by whylabs_client's enums. but maybe it's best to build an enum on whylabs_toolkit too